src/rdkafka_mock_cgrp.c

```diff
@@ -1402,19 +1402,65 @@ rd_kafka_mock_cgrp_consumer_member_find_by_instance_id(
     rd_kafka_mock_cgrp_consumer_member_t *member,
     rd_kafkap_str_t *SubscribedTopicNames,
-    int32_t SubscribedTopicNamesCnt) {
-        rd_bool_t changed = rd_false;
-        if (!SubscribedTopicNames)
+    int32_t SubscribedTopicNamesCnt,
+    const rd_kafkap_str_t *SubscribedTopicRegex) {
+        rd_bool_t changed           = rd_false;
+        rd_list_t *new_subscription = NULL;
+        int32_t i;
+
+        if (!SubscribedTopicNames &&
+            RD_KAFKAP_STR_IS_NULL(SubscribedTopicRegex) &&
+            !member->subscribed_topic_regex)
                 /* When client is sending NULL,
                  * its subscription didn't change */
                 return changed;
 
-        int32_t i;
-        rd_list_t *new_subscription = rd_list_new(
-            SubscribedTopicNamesCnt > 0 ? SubscribedTopicNamesCnt : 1, rd_free);
-        for (i = 0; i < SubscribedTopicNamesCnt; i++) {
-                rd_list_add(new_subscription,
+        if (!RD_KAFKAP_STR_IS_NULL(SubscribedTopicRegex)) {
+                if (member->subscribed_topic_regex) {
+                        rd_kafkap_str_destroy(member->subscribed_topic_regex);
+                        member->subscribed_topic_regex = NULL;
+                }
+                if (SubscribedTopicRegex->len > 0)
+                        member->subscribed_topic_regex =
+                            rd_kafkap_str_copy(SubscribedTopicRegex);
+        }
+
+        if (member->subscribed_topic_regex) {
+                rd_kafka_mock_cluster_t *mcluster = member->mcgrp->cluster;
+                rd_kafka_mock_topic_t *mtopic;
+                char errstr[1];
+                rd_regex_t *re =
+                    rd_regex_comp(member->subscribed_topic_regex->str, errstr,
+                                  sizeof(errstr));
+
+                TAILQ_FOREACH(mtopic, &mcluster->topics, link) {
+                        if (rd_regex_exec(re, mtopic->name)) {
+                                rd_kafkap_str_t *topic_name =
+                                    rd_kafkap_str_new(mtopic->name, -1);
+                                if (!new_subscription)
+                                        new_subscription =
+                                            rd_list_new(1, rd_free);
+                                rd_list_add(new_subscription,
+                                            RD_KAFKAP_STR_DUP(topic_name));
+                                rd_kafkap_str_destroy(topic_name);
+                        }
+                }
+
+                rd_regex_destroy(re);
+        }
+
+        if (SubscribedTopicNames) {
+                if (!new_subscription)
+                        new_subscription =
+                            rd_list_new(SubscribedTopicNamesCnt > 0
+                                            ? SubscribedTopicNamesCnt
+                                            : 1,
+                                        rd_free);
```

Comment:
we can remove this when creating it in advance. Use `SubscribedTopicNamesCnt` there too.